infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -53,12 +67,23 @@ public KubernetesNamespaceFactory(
       @Nullable @Named("che.infra.kubernetes.namespace") String namespaceName,
       @Nullable @Named("che.infra.kubernetes.service_account_name") String serviceAccountName,
       @Nullable @Named("che.infra.kubernetes.cluster_role_name") String clusterRoleName,
-      KubernetesClientFactory clientFactory) {
+      @Named("che.infra.kubernetes.namespace.default") String defaultNamespaceName,
+      @Named("che.infra.kubernetes.namespace.allow_user_defined")
+          boolean allowUserDefinedNamespaces,
+      KubernetesClientFactory clientFactory)
+      throws ConfigurationException {
     this.namespaceName = namespaceName;
     this.isPredefined = !isNullOrEmpty(namespaceName) && hasNoPlaceholders(this.namespaceName);
     this.serviceAccountName = serviceAccountName;
     this.clusterRoleName = clusterRoleName;
     this.clientFactory = clientFactory;
+    this.defaultNamespaceName = defaultNamespaceName;
+    this.allowUserDefinedNamespaces = allowUserDefinedNamespaces;
+    if (isNullOrEmpty(defaultNamespaceName) && !allowUserDefinedNamespaces) {
```

Comment:
If `defaultNamespaceName` cannot be null, this check should be replaced with `isEmpty()`

Reply:
now it can